pages/index.js

```diff
@@ -174,34 +192,43 @@ const ConfirmationDialog = styled.div`
   }
 `;
 
-function useLocalStorageState(key, defaultValue) {
-  const [state, setState] = useState(defaultValue);
-
-  useEffect(() => {
-    if (typeof window !== "undefined") {
-      const storedValue = localStorage.getItem(key);
-      if (storedValue) {
-        setState(JSON.parse(storedValue));
-      }
-    }
-  }, [key]);
-
-  useEffect(() => {
-    if (typeof window !== "undefined") {
-      localStorage.setItem(key, JSON.stringify(state));
-    }
-  }, [key, state]);
-
-  return [state, setState];
-}
-
 const ShoppingList = () => {
   const [items, setItems] = useLocalStorageState("shoppingItems", []);
+  const [purchasedItems, setPurchasedItems] = useLocalStorageState(
+    "purchasedItems",
+    []
+  );
   const [errors, setErrors] = useState({});
   const [deletingItemId, setDeletingItemId] = useState(null);
 
+  const totalItems = (items || []).reduce(
+    (total, item) => total + item.quantity,
+    0
+  );
+
+  const TotalPurchased = (purchasedItems || []).reduce(
```

Comment:
the .reduced could directly be applied to the purchasedItems as it's not necessary to use it on an empty array